src/Combinatorics/Matroids/matroids.jl

```diff
@@ -0,0 +1,769 @@
+```jldoctest
+julia> B = [[1,2],[1,3],[1,4],[2,3],[2,4]];
+
+julia> M = matroid_from_bases(B,4)
+Matroid of rank 2 on 4 elements
+```
+
+# Examples
+To construct the same matroid on the four elements 1,2,i,j you may write:
+```jldoctest
+julia> M = matroid_from_bases([[1,2],[1,'i'],[1,'j'],[2,'i'],[2,'j']],[1,2,'i','j'])
+Matroid of rank 2 on 4 elements
+```
+"""
+matroid_from_bases(bases::Union{AbstractVector{<:AbstractVector{<:IntegerUnion}}, AbstractVector{<:AbstractSet{<:IntegerUnion}}}, nelements::IntegerUnion) = matroid_from_bases(bases,Vector(1:nelements))
+
+function matroid_from_bases(bases::Union{AbstractVector{<:AbstractVector}, AbstractVector{<:AbstractSet}},groundset::AbstractVector; check::Bool=true)
+    if check && length(groundset)!=length(Set(groundset))
+        error("Input is not a valid groundset of a matroid")
+    end
+    gs2num = Dict{Any,IntegerUnion}()
+    i = 1
+    for elem in groundset
+        gs2num[elem] = i
+        i+=1
+    end
+    pm_bases = [[gs2num[i]-1 for i in B] for B in bases]
+    M = Polymake.matroid.Matroid(BASES=pm_bases,N_ELEMENTS=length(groundset))
+    if check && !Polymake.matroid.check_basis_exchange_axiom(M.BASES)
+        error("Input is not a collection of bases")
+    end
+    return Matroid(M,groundset,gs2num)
+end
+
+@doc Markdown.doc"""
+Construct a `matroid` with nonbases `N` on the ground set `E` (which can be the empty set).
+That means that the matroid has as bases all subsets of the size `|N[1]|` of the ground set that are not in `N`.
+There `N` can't be empty.
+The described complement of `N` needs to be a non-empty collection of subsets of the ground set `E` satisfying an exchange property,
+and the default value for `E` is the set `{1,..n}` for a non-negative value `n`.
+
+See Section 1.2 of Oxl11 (@cite)
+
+# Examples
+To construct the Fano matroid you may write:
+```jldoctest
+julia> H = [[1,2,4],[2,3,5],[1,3,6],[3,4,7],[1,5,7],[2,6,7],[4,5,6]];
+
+julia> M = matroid_from_nonbases(H,7)
+Matroid of rank 3 on 7 elements
+
+```
+"""
+matroid_from_nonbases(nonbases::Union{AbstractVector{<:AbstractVector{<:IntegerUnion}}, AbstractVector{<:AbstractSet{<:IntegerUnion}}}, nelements::IntegerUnion) = matroid_from_nonbases(nonbases,Vector(1:nelements))
+
+function matroid_from_nonbases(nonbases::Union{AbstractVector{<:AbstractVector}, AbstractVector{<:AbstractSet}},groundset::AbstractVector; check::Bool=true)
+    if check && length(groundset)!=length(Set(groundset))
+        error("Input is not a valid groundset of a matroid")
+    end
+    gs2num = Dict{Any,IntegerUnion}()
+    i = 1
+    for elem in groundset
+        gs2num[elem] = i
+        i+=1
+    end
+    if length(nonbases)==0
+        error("The collection of nonbases should not be empty.")
+    end
+    rk = length(nonbases[1])
+    bases = Vector{Vector{IntegerUnion}}()
+    for set in Oscar.Hecke.subsets(Vector(1:length(groundset)),rk)
+        if !(set in nonbases)
+            push!(bases, set)
+        end
+    end
```

Comment:
Can this block be rewritten as some variant of `filter`? Please look below for an example.

Reply:
Done.